python/cugraph/cugraph/sampling/heterogeneous_neighbor_sample.py

```diff
@@ -0,0 +1,349 @@
+                    Contains the offsets of each batch in the sampling result
+
+                If renumber=True:
+                    (adds the following dataframe)
+                    renumber_df['renumber_map']: cudf.Series
+                        Contains the renumber maps for each batch
+                    renumber_df['batch_id']: cudf.Series
+                        Contains the batch ids for the renumber maps
+
+    """
+
+
+    if compression not in ["COO", "CSR", "CSC", "DCSR", "DCSC"]:
+        raise ValueError("compression must be one of COO, CSR, CSC, DCSR, or DCSC")
+
+    if (
+        (compression != "COO")
+        and (not compress_per_hop)
+        and prior_sources_behavior != "exclude"
+    ):
+        raise ValueError(
+            "hop-agnostic compression is only supported with"
+            " the exclude prior sources behavior due to limitations "
+            "of the libcugraph C++ API"
+        )
+
+    if compress_per_hop and prior_sources_behavior != "carryover":
+        raise ValueError(
+            "Compressing the edgelist per hop is only supported "
+            "with the carryover prior sources behavior due to limitations"
+            " of the libcugraph C++ API"
+        )
+
+    if isinstance(start_list, int):
+        start_list = [start_list]
+
+    if isinstance(start_list, list):
+        start_list = cudf.Series(
+            start_list, dtype=G.edgelist.edgelist_df[G.srcCol].dtype
+        )
+    
+    if isinstance(starting_vertex_label_offsets, list):
+        starting_vertex_label_offsets = cudf.Series(
+            starting_vertex_label_offsets
+        )
+    
+    if isinstance(vertex_type_offsets, list):
+        vertex_type_offsets = cudf.Series(
+            vertex_type_offsets
+        )
+
+    # fanout_vals must be passed to pylibcugraph as a host array
+    if isinstance(fanout_vals, numpy.ndarray):
+        fanout_vals = fanout_vals.astype("int32")
+    elif isinstance(fanout_vals, list):
+        fanout_vals = numpy.asarray(fanout_vals, dtype="int32")
+    elif isinstance(fanout_vals, cp.ndarray):
+        fanout_vals = fanout_vals.get().astype("int32")
+    elif isinstance(fanout_vals, cudf.Series):
+        fanout_vals = fanout_vals.values_host.astype("int32")
+    else:
+        raise TypeError("fanout_vals must be a sequence, " f"got: {type(fanout_vals)}")
+
+    if "weights" in G.edgelist.edgelist_df:
+        weight_t = G.edgelist.edgelist_df["weights"].dtype
+    else:
+        weight_t = "float32"
+
+    start_list = ensure_valid_dtype(G, start_list)
+
+    if G.renumbered is True:
+        if isinstance(start_list, cudf.DataFrame):
+            start_list = G.lookup_internal_vertex_id(
+                start_list, start_list.columns
+            )
+        else:
+            start_list = G.lookup_internal_vertex_id(start_list)
+
+    if with_biases:
+        sampling_result_array_dict = pylibcugraph_heterogeneous_biased_neighbor_sample(
```

Comment:
Are the calls exactly the same except for the PLC function itself? If so, you could trim this down quite a bit like this:
```suggestion
    if with_biases:
        sampling_function = pylibcugraph_heterogeneous_biased_neighbor_sample
    else:
        sampling_function = pylibcugraph_heterogeneous_uniform_neighbor_sample
```
Then just call it once as 
```
    sampling_result_array_dict = sampling_function( ... )
```

Reply:
That will work for now since all arguments are the same however, once we start supporting `biases` input argument for `biased_neighbor_sample`, it won't.